client/js/main.js

```diff
@@ -0,0 +1,199 @@
+'use strict';
```

Comment:
Separate all in proper pieces. You should clearly understand which entities you have. Task is a task, while main is something different. It probably only needs to initialize some stuff inside.